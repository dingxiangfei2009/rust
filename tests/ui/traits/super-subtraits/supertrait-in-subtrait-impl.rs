#![feature(rustc_attrs)]

trait Sup {
    type A;
}

#[rustc_supertrait_in_subtrait_impl]
trait Sub: Sup {}

struct S;
impl Sub for S {
    //~^ ERROR: the trait bound `S: Sup` is not satisfied
    type A = ();
    //~^ ERROR: the trait bound `S: Sup` is not satisfied
}

fn main() {}
