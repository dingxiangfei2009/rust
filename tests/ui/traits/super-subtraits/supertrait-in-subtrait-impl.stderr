error[E0277]: the trait bound `S: Sup` is not satisfied
  --> $DIR/supertrait-in-subtrait-impl.rs:13:14
   |
LL |     type A = ();
   |              ^^ the trait `Sup` is not implemented for `S`
   |
help: this trait has no implementations, consider adding one
  --> $DIR/supertrait-in-subtrait-impl.rs:3:1
   |
LL | trait Sup {
   | ^^^^^^^^^

error[E0277]: the trait bound `S: Sup` is not satisfied
  --> $DIR/supertrait-in-subtrait-impl.rs:11:14
   |
LL | impl Sub for S {
   |              ^ the trait `Sup` is not implemented for `S`
   |
help: this trait has no implementations, consider adding one
  --> $DIR/supertrait-in-subtrait-impl.rs:3:1
   |
LL | trait Sup {
   | ^^^^^^^^^
note: required by a bound in `Sub`
  --> $DIR/supertrait-in-subtrait-impl.rs:8:12
   |
LL | trait Sub: Sup {}
   |            ^^^ required by this bound in `Sub`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
